components/ShoppingItem/ShoppingItem.js

```diff
@@ -1,37 +1,53 @@
-import styled from "styled-components";
+import styled, { ThemeConsumer } from "styled-components";
 import Link from "next/link";
 import { useState } from "react";
 
 export default function ShoppingItem({
-  id,
   onDeleteItem,
   shoppingItem,
   onEditItem,
   onChangeMode,
-  mode,
+  onToggleIsPurchased,
+  purchasedItems,
 }) {
   const [isToBeDeleted, setIsToBeDeleted] = useState(false);
 
-  // const [isToBeEdited, setIsToBeEdited] = useState(false);
-
   function toggleIsToBeDeleted() {
     setIsToBeDeleted(!isToBeDeleted);
   }
 
-  // function toggleIsToBeEdited() {
-  //   setIsToBeEdited(!isToBeEdited);
-  // }
-
   return (
     <ListItem
-      key={shoppingItem.id}
       $backgroundColor={shoppingItem.backgroundColor}
+      $isPurchased={shoppingItem.isPurchased}
     >
       <CategoryBox>{shoppingItem.category}</CategoryBox>
-
       {!isToBeDeleted ? (
         <>
-          {shoppingItem.name}: {shoppingItem.quantity}
+          <StyledTickboxNameNumberContainer>
+            <form>
+              <input
+                type="checkbox"
+                id={`checkbox-${shoppingItem.id}`}
+                checked={shoppingItem.isPurchased}
+                onChange={() => {
+                  onToggleIsPurchased(shoppingItem.id);
+                }}
+              />
+              <label htmlFor={`checkbox-${shoppingItem.id}`}>
+                {purchasedItems.includes(shoppingItem.id) ? "purchased" : ""}
+              </label>
+            </form>
```

Comment:
In this case you do not really need a form. In order to toggle the label of the checkbox, you can use the `isPurchased` key of your `shoppingItem`:

```suggestion
              <input
                type="checkbox"
                id={`checkbox-${shoppingItem.id}`}
                checked={shoppingItem.isPurchased}
                onChange={() => {
                  onToggleIsPurchased(shoppingItem.id);
                }}
              />
              <label htmlFor={`checkbox-${shoppingItem.id}`}>
              {shoppingItem.isPurchased ? "purchased" : ""}
            </label>
```

Reply:
As deleting the <form> element changes the arrangement on the screen, I leave it in for now and will consider removing it in the Design US.

Reply:
You could use another container element here, if you like.